server/src/main/java/org/opensearch/cluster/routing/remote/InternalRemoteRoutingTableService.java

```diff
@@ -273,6 +310,66 @@ private void uploadIndex(
         }
     }
 
+    private void uploadIndexRoutingDiff(
+        String clusterUUID,
+        Map<String, Diff<IndexRoutingTable>> indexRoutingTableDiff,
+        String fileName,
+        BlobContainer blobContainer,
+        ActionListener<Void> completionListener
+    ) {
+        BytesReference bytesInput = null;
+        try (BytesStreamOutput streamOutput = new BytesStreamOutput()) {
+            RemoteIndexRoutingTableDiff remoteIndexRoutingTableDiff = new RemoteIndexRoutingTableDiff(
+                indexRoutingTableDiff,
+                clusterUUID,
+                compressor
+            );
+            remoteIndexRoutingTableDiff.writeTo(streamOutput);
+            bytesInput = streamOutput.bytes();
+        } catch (IOException e) {
+            logger.error(() -> "Failed to serialize IndexRoutingTableDiff: " + e.getMessage(), e);
+            completionListener.onFailure(e);
+            return;
+        }
+
+        if (!(blobContainer instanceof AsyncMultiStreamBlobContainer)) {
+            try {
+                blobContainer.writeBlob(fileName, bytesInput.streamInput(), bytesInput.length(), true);
```

Comment:
The writeblob has different flow for bigger file sizes. Can we add tests with > 10Mb and see everything is working as expected?

Reply:
Can we track it in separate issue? I think we can take it up separate, need not be a part of this PR.